### PR TITLE
Create programming answer annotations

### PR DESCRIPTION
**Pull request by lowjoel**
- Create associated models
 - Implement explicit line numbering when formatting code blocks
 - Implement the UI for creating new annotations in programming answers

Next up will be creating the UI for displaying the existing annotations.

### CODE REVIEW

File: app/assets/javascripts/course/assessment/submission/answer/programming.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+  $(document).on('DOMNodeInserted', function(e) {
+    addProgrammingAnnotationLinks(e.target);
+  });
+  $(document).on('click', DOCUMENT_SELECTOR + 'table.codehilite .add-annotation',
+    onAddProgrammingAnnotation);
+  $(document).on('click', DOCUMENT_SELECTOR + '.annotation-form input[type="reset"]',
+    onAnnotationFormResetted);
+  $(document).on('click', DOCUMENT_SELECTOR + '.annotation-form input[type="submit"]',
+    onAnnotationFormSubmitted);
+})(jQuery);
```
Comment (lowjoel) [inline app/assets/javascripts/course/assessment/submission/answer/programming.js]: Obviously Hound is ignoring our jshint config.

File: app/assets/javascripts/course/assessment/submission/answer/programming.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+   * This is a less sophisticated version of $.serialize() in that it only supports inputs and
+   * textareas.
+   *
+   * @param {jQuery} $form The form being submitted.
+   * @returns {Object} The form data to be submitted.
+   */
+  function buildFormData($form) {
+    var $fields = findFormFields($form, ':not(:disabled)');
+    var data = {
+      authenticity_token: $(document).find('meta[name="csrf-token"]').attr('content')
```
Comment (lowjoel) [inline app/assets/javascripts/course/assessment/submission/answer/programming.js]: This is needed for the CSRF token.

**Comment by allenwq** [comment]
@lowjoel Looks good ! but the build failed.

**Comment by lowjoel** [comment]
I know =( can't figure out why

@lowjoel<https://github.com/lowjoel> Looks good ! but the build failed.

**Pull request merged**